modules/external-auth-configuring.adoc

```diff
@@ -0,0 +1,192 @@
+      componentNamespace: openshift-console
+  type: OIDC <11>
+  webhookTokenAuthenticator: null <12>
+----
+<1> The OIDC provider configuration.
+<2> The name of the claim to construct group names for the cluster identity.
+<3> The name of the claim to construct usernames for the cluster identity.
+<4> The list of audience IDs that this authentication provider issues tokens for.
+<5> The name of the config map that contains the `ca-bundle.crt` key. If unset, system trust is used instead.
+<6> The URL for the token issuer.
+<7> The name for external OIDC provider.
+<8> The client ID that your provider uses for the {oc-first}.
+<9> The client ID that your provider uses for the {product-title} web console.
+<10> The name of the secret that stores the secret value for the console client.
+<11> A value of `OIDC` indicates to use an external OIDC identity provider.
+<12> Must be set to `null` when `type` is set to `OIDC`.
++
+//TODO what does end of #5 mean? https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-79807 - don't see the difference in config for issuerCertificateAuthority being unset
++
+For more details on all available parameters, see "OIDC provider configuration parameters".
+
+. Exit and save the changes to apply the new configuration.
+
+.Verification
+
+. Wait for the cluster to roll out new revisions to all nodes.
+
+.. Check the Kubernetes API server Operator status by running the following command:
++
+[source,terminal]
+----
+$ oc get co kube-apiserver
+----
++
+.Example output
+[source,terminal]
+----
+NAME             VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
+kube-apiserver   4.19.0    True        True          False      85m     NodeInstallerProgressing: 2 node are at revision 8; 1 node is at revision 10
+----
++
+The message in the above example shows that one node has progressed to the new revision and two nodes have not yet updated. It can take 30 minutes or more to roll out the new revision to all nodes, depending on the size of your cluster.
++
+// .. TODO: Check anything else, or is the above sufficient?
+
+. Verify that you can log in to the {oc-first} by authenticating with Keycloak:
+
+.. Run the `oc login` command by running the following command:
++
+[source,terminal]
+----
+$ oc login --exec-plugin=oc-oidc --issuer-url=https://keycloak-keycloak.apps.example.com/realms/master \// <1>
+    --client-id=oc-cli-test \// <2>
+    --extra-scopes=email --callback-port=8080 \
+    --oidc-certificate-authority my-directory/ca-bundle.crt <3>
+----
+<1> Specify the issuer URL for your Keycloak provider.
+<2> Specify client ID for the {oc-first}.
+<3> Specify the path to the `ca-bundle.crt` file on your local machine.
++
+.Example output
+[source,terminal]
+----
+WARNING: Using insecure TLS client config. Setting this option is not supported!
+
+Please visit the following URL in your browser: http://localhost:8080
+----
+
+.. Open http://localhost:8080 in a browser.
```

Comment:
@liouk @xingxingxia Will it always be localhost:8080 no matter their keycloak configuration?

Reply:
As replied in the client question, it consists with the redirect uris set there, as the test case step.